packages/SwingSet/src/controller/startXSnap.js

```diff
@@ -5,6 +5,32 @@ import { xsnap, recordXSnap } from '@agoric/xsnap';
 
 const NETSTRING_MAX_CHUNK_SIZE = 12_000_000;
 
+/**
+ * @typedef {object} StartXSnapLoadFromBundleDetails
```

Comment:
There are 2 high level cases:
- start xsnap process from scratch and send command to load some bundles
- start xsnap process from snapshot (which really should be start from scratch and send command to load a snapshot, but that's an improvement for another day)

That's what `options.load` attempts to capture. The other top level options for `startXSnap` should be independent of how the load is performed, which means the `bundleIDs` have no reason to appear in the top-level `startXSnap` options if we're loading from a snapshot. I did not have time to change the plumbing for the load from bundle path, but the `bundleIDs` should move under the `options.load = { source: 'bundle', bundlesIDs };`.

> Although.. `startXSnap` is not the piece that handles vat bundles, they need to be sent as a command payload, after the supervisor is running (they must be evaluated by liveslots

I am confused, from what I gather, the `bundleIDs` are the lockdown and supervisor bundles. This file does not deal with the vat's source bundle at all from what I can tell.

Reply:
Ah, ok, the singular "bundle" in the old `StartXSnapLoadFromBundleDetails` name made me think the (singular) vat bundle was involved, and that was at odds with the "load" verb (which I associate with snapshots). I definitely agree that this first "from scratch" case is the only one where the helper bundleIDs (plural) should appear.

And now I see that you're aiming for a single argument which captures both the "from scratch (with helper bundles)" case and the "from snapshot (either 'here is a stream' or 'go ask snapstore')" case, where I was thinking about separate arguments, so finding a single name for the combined one (and a union type) is awkward.

Let's not change it now, but in a later cleanup, let's consider splitting `startXSnap` into two functions, one for from-scratch, the other for load-snapshot: I've been thinking about doing the same at the manager-subprocess-xsnap.js and vat-warehouse.js layers. The vat-warehouse.js `ensureVatOnline` will look at the first entry in the transcript and then either vector off into the "from scratch" path (using `initOptions`, which includes `workerOptions` and the helper bundleIDs), or the "from snapshot" path.

Reply:
I'm wondering if we should consider always starting a blank worker, then a command to load bundle, or another command to load a snapshot. At least at the xsnap layer it would make sense.